File: MetaSwapV3/sdk/node/metaswap-client.js
export class MetaSwapClient {
  constructor({ baseUrl, apiKey }) {
    this.baseUrl = baseUrl.replace(/\/$/, "");
    this.apiKey = apiKey;
  }

  health() {
    return this.get("/health");
  }

  plans() {
    return this.get("/developer/plans");
  }

  revenueScalePlan() {
    return this.get("/revenue/scale-plan");
  }

  rpc({ chain, method, params = [] }) {
    return this.post("/rpc/proxy", { chain, method, params });
  }

  accelerationQuote({ chain, urgency = "standard", estimatedGas = 21000 }) {
    return this.post("/tx/acceleration/quote", { chain, urgency, estimatedGas });
  }

  relayTransaction({ chain, rawTransaction, urgency = "standard" }) {
    return this.post("/tx/relay", { chain, rawTransaction, urgency, userConsent: true });
  }

  subscribeWebhook({ url, events }) {
    return this.post("/webhooks", { url, events });
  }

  async get(path) {
    return this.request("GET", path);
  }

  async post(path, body) {
    return this.request("POST", path, body);
  }

  async request(method, path, body) {
    const response = await fetch(`${this.baseUrl}${path}`, {
      method,
      headers: {
        "content-type": "application/json",
        "x-api-key": this.apiKey
      },
      body: body ? JSON.stringify(body) : undefined
    });
    const payload = await response.json();
    if (!response.ok) throw new Error(payload.error ?? `HTTP ${response.status}`);
    return payload;
  }
}

File: MetaSwapV3/sdk/python/metaswap_client.py
import requests


class MetaSwapClient:
    def __init__(self, base_url: str, api_key: str | None = None):
        self.base_url = base_url.rstrip("/")
        self.api_key = api_key

    def health(self):
        return self.get("/health")

    def plans(self):
        return self.get("/developer/plans")

    def revenue_scale_plan(self):
        return self.get("/revenue/scale-plan")

    def rpc(self, chain: str, method: str, params=None):
        return self.post("/rpc/proxy", {"chain": chain, "method": method, "params": params or []})

    def acceleration_quote(self, chain: str, urgency: str = "standard", estimated_gas: int = 21000):
        return self.post("/tx/acceleration/quote", {"chain": chain, "urgency": urgency, "estimatedGas": estimated_gas})

    def relay_transaction(self, chain: str, raw_transaction: str, urgency: str = "standard"):
        return self.post("/tx/relay", {"chain": chain, "rawTransaction": raw_transaction, "urgency": urgency, "userConsent": True})

    def subscribe_webhook(self, url: str, events: list[str]):
        return self.post("/webhooks", {"url": url, "events": events})

    def get(self, path: str):
        return self.request("GET", path)

    def post(self, path: str, body: dict):
        return self.request("POST", path, body)

    def request(self, method: str, path: str, body: dict | None = None):
        headers = {"content-type": "application/json"}
        if self.api_key:
            headers["x-api-key"] = self.api_key
        response = requests.request(method, f"{self.base_url}{path}", json=body, headers=headers, timeout=30)
        response.raise_for_status()
        return response.json()
